### DISCUSSION TITLE
Raspberry Pi 5 Announced

### DISCUSSION THREAD

**Discussion by THX723** [marcone/teslausb#780] [General]
Just announced is the new RPi 5. Availability is sometime next month (October). Pre-orders happening as we speak.

https://www.dexerto.com/tech/raspberry-pi-5-finally-revealed-alongside-major-upgrades-2313650/

More in-depth tech details is needed, but the new addition of M.2 SSD support is interesting and hopefully the previous USB3 design flaw with OTG/gadget-mode support has been addressed.

We shall see.

**Comment by THX723** [comment]
Dug into the USB tidbit more...

It looks like RPi 5 introduced a new architecture with the introduction of an in-house RP1 southbridge to handles the I/O. This means no more VLI VL805 USB controller, which was partially the culprit of no OTG support for the USB3 port. This is encouraging news, but not yet a done deal. We'll know more as additional details on the RP1 wrt USB3 trickles out.

**Comment by THX723** [comment]
Looks like a dead-end. Same USB 2.0 limitation for the USB-C port that support OTG. Bummer!

**Reply by marcone** [reply]
Where did you find this info? From what little I've been able to find it seemed to me that the USB-C port is now just a power port, and all USB goes through this new chip that handles the two USB3 and two USB2 ports.

**Reply by THX723** [reply]
Across a few places...

> https://www.jeffgeerling.com/blog/2023/answering-some-questions-about-raspberry-pi-5
> https://community.element14.com/products/raspberry-pi/f/raspberry-pi-5-faq/53736/does-the-raspberry-pi-5-support-otg

and also in the official raspberry pi forum. This is a big thread, which requires you to hop around more for OTG related conversation.

> https://forums.raspberrypi.com/viewtopic.php?p=2139306&hilit=otg#p2139306

Admiringly, none of the sources provided hard evidence, so just taking their answers for what it is at this time.

**Reply by THX723** [reply]
another source from the Official Raspberry Pi 5 introduction page comment section by a staff member:

> https://www.raspberrypi.com/news/introducing-raspberry-pi-5/
> Raspberry Pi Staff  Gordon Hollingworth
> 28th September 2023, 10:00 am
> The USB-C connector has the same dwc-otg interface capability as Pi 4. It will also go into rpiboot mode if you hold down the power button when you power it.

**Reply by THX723** [reply]
> Jarom Hatch
> 28th September 2023, 9:37 am
> Does the USB-C port have USB3 host support now rather than USB2?
> 
> Raspberry Pi Staff  Gordon Hollingworth
> 28th September 2023, 10:28 am
> No, the USB-C is the same dwc-otg peripheral as before, it’s the most reliable one we have (we’ve now been using it for 12 years)…

**Reply by marcone** [reply]
That's disappointing. I was really hoping for feature parity with the newer Rockchip devices, but with the better support from Raspberry Pi.

**Reply by THX723** [reply]
Agreed. I suppose OTG use cases aren't that popular, so it was never on their (or anyones) feature development roadmap. What we get and are stuck with is simply the baseline/legacy feature built into the original Broadcom design. Raspberry Pi Foundation had an opportunity when they invested millions developing their custom southbridge RP1 chip. A missed opportunity!
